File: Serene/Serene.Script/Administration/Role/RoleDialog.cs

using Serenity;
using System.Runtime.CompilerServices;

namespace Serene.Administration
{
    [Imported]
    public class RoleDialog : EntityDialog<RoleRow>
    {
    }

    /*
    This class has been ported to TypeScript. See RoleDialog.ts
    Code below is only for reference purposes who wants to use Saltaralle 


    [IdProperty("RoleId"), NameProperty("RoleName")]
    [FormKey("Administration.Role"), LocalTextPrefix("Administration.Role"), Service("Administration/Role")]
    public class RoleDialog : EntityDialog<RoleRow>, IAsyncInit
    {
        protected override List<ToolButton> GetToolbarButtons()
        {
            var buttons = base.GetToolbarButtons();

            buttons.Add(new ToolButton
            {
                Title = Q.Text("Site.RolePermissionDialog.EditButton"),
                CssClass = "lock-button",
                OnClick = delegate
                {
                    new RolePermissionDialog(new RolePermissionDialogOptions
                    {
                        RoleID = this.Entity.RoleId.Value,
                        Title = this.Entity.RoleName,
                    }).DialogOpen();
                }
            });

            return buttons;
        }

        protected override void UpdateInterface()
        {
            base.UpdateInterface();

            toolbar.FindButton("lock-button").ToggleClass("disabled", this.IsNewOrDeleted);
        }
    }
    */
}

File: Serene/Serene.Web/Modules/Administration/Role/RoleDialog.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    export class RoleDialog extends Serenity.EntityDialog<RoleRow, any> {
        protected getFormKey() { return RoleForm.formKey; }
        protected getIdProperty() { return RoleRow.idProperty; }
        protected getLocalTextPrefix() { return RoleRow.localTextPrefix; }
        protected getNameProperty() { return RoleRow.nameProperty; }
        protected getService() { return RoleService.baseUrl; }

        protected form = new RoleForm(this.idPrefix);

        protected getToolbarButtons()
        {
            let buttons = super.getToolbarButtons();

            buttons.push({
                title: Q.text('Site.RolePermissionDialog.EditButton'),
                cssClass: 'edit-permissions-button',
                icon: 'icon-lock-open text-green',
                onClick: () =>
                {
                    new RolePermissionDialog({
                        roleID: this.entity.RoleId,
                        title: this.entity.RoleName
                    }).dialogOpen();
                }
            });

            return buttons;
        }

        protected updateInterface() {
            super.updateInterface();

            this.toolbar.findButton("edit-permissions-button").toggleClass("disabled", this.isNewOrDeleted());
        }
    }
}